src/sentry/search/events/builder.py

```diff
@@ -1900,6 +1900,13 @@ def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[Whe
         # If this is an aliasedexpression, we don't need the alias here, just the expression
         if isinstance(lhs, AliasedExpression):
             lhs = lhs.exp
+        # Special case transaction or title since they're actually transforms
+        if is_txn_query := (
```

Comment:
nit: Should we add transaction filter logic to the `search_filter_converter` so we're not overloading the default filter with metrics dataset specific logic? since the transform logic we're checking for is special to the metrics dataset?